src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -485,7 +485,7 @@ private function createAutowiredDefinition($type)
     private function createTypeNotFoundMessage(TypedReference $reference, $label)
     {
         if (!$r = $this->container->getReflectionClass($type = $reference->getType(), false)) {
-            $message = sprintf('has type "%s" but this class cannot be loaded.', $type);
+            $message = sprintf('is type-hinted with "%s" but this class was not found.', $type);
```

Comment:
The "cannot be loaded" case is possible when the parent class is not found. Maybe and "or the/a parent class is not found"? (Not sure about the wording but that's explaining the current one.)

Reply:
Is there any issue with addition a conditional?

if (class_exists($type)) {
    // we know a parent class must not exist
} else {
    // we know that the class itself does not exist
}

Reply:
`class_exists($type)` would trigger a fatal error when the parent is missing. `container->getReflectionClass()` contains a protection for that situation. Maybe passing "true" as 2nd argument to getReflectionClass and dealing with the exception could help make the message better?

Reply:
also please don't use type "hints". those are not hints like phpdocs.

Reply:
"hint" is used everywhere, in Symfony, in the PHP doc, on google. Technically is a bit more than a "hint" I agree, but that's what ppl are used to now IMHO.

Reply:
Not at all. It was explicitly avoided to call the hints in the php doc. See http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration
And also the RFC about type declarations does not use hints for this reason.